KiedyKolos/Common/KiedyKolos.Core/Validators/Group/UpdateYearCourseGroupCommandValidator.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentValidation;
+using KiedyKolos.Core.Commands;
+using KiedyKolos.Core.Interfaces;
+
+namespace KiedyKolos.Core.Validators.Group
+{
+    public class UpdateYearCourseGroupCommandValidator : AbstractValidator<UpdateYearCourseGroupCommand>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public UpdateYearCourseGroupCommandValidator(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+
+            RuleFor(g => g).MustAsync(async (g, cancellation) =>
+            {
+                var yearCourse = await _unitOfWork.YearCourseRepository.GetAsync(g.YearCourseId);
+                return yearCourse != null;
+            }).WithMessage("Given year course does not exist");
+
+            RuleFor(g => g).MustAsync(async (g, cancellation) =>
+            {
+                var yearCourse = await _unitOfWork.YearCourseRepository.GetAsyncWithGroups(g.YearCourseId);
+                return yearCourse.Groups.Any(ycg => ycg.Id == g.GroupId);
+            }).WithMessage("Given year course group does not exist");
+
+            RuleFor(g => g).MustAsync(async (g, cancellation) =>
+            {
+                var yearCourse = await _unitOfWork.YearCourseRepository.GetAsync(g.YearCourseId);
+                return !yearCourse.Groups.Any(ycg => ycg.GroupNumber == g.GroupNumber || ycg.GroupName == g.GroupName);
```

Comment:
LINQ expression should be checking if group with same number and name isn't the same group(validation occurs before update and transaction, user could update group with the same name and the same number as it already has, and that should pass).